### ISSUE TITLE
"Not enough space in the context's memory pool"

### ISSUE THREAD

**Issue by TheBloke** [marella/gpt4all-j#2]
Hi

When trying longer prompts, Python crashes with `not enough space in the context's memory pool`.  Example:

```
tomj@Eddie ~/src $ python
Python 3.11.3 (main, Apr  7 2023, 19:25:52) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from gpt4allj import Model
>>> model = Model('/Users/tomj/src/test-gpt4all/models/ggml-gpt4all-j-v1.2-jazzy.bin')
gptj_model_load: loading model from '/Users/tomj/src/test-gpt4all/models/ggml-gpt4all-j-v1.2-jazzy.bin' - please wait ...
gptj_model_load: n_vocab = 50400
gptj_model_load: n_ctx   = 2048
gptj_model_load: n_embd  = 4096
gptj_model_load: n_head  = 16
gptj_model_load: n_layer = 28
gptj_model_load: n_rot   = 64
gptj_model_load: f16     = 2
gptj_model_load: ggml ctx size = 5401.45 MB
gptj_model_load: memory_size =  1792.00 MB, n_mem = 57344
gptj_model_load: ................................... done
gptj_model_load: model size =  3609.38 MB / num tensors = 285
>>> prompt='''Below is an instruction that describes a task. Write a response that appropriately completes the request.
... ### Instruction:
... Write a story about llamas
... ### Response:'''
>>> print(model.generate(prompt))
gptj_generate: seed = 1681815129
gptj_generate: number of tokens in prompt = 34

Below is an instruction that describes a task. Write a response that appropriately completes the request.
### Instruction:
Write a story about llamas
### Response:ggml_new_tensor_impl: not enough space in the context's memory pool (needed 1382304912, available 612697122)
[1]    8718 segmentation fault  python
```

**Comment by marella** [comment]
There was a bug where I'm not passing n_batch param which was fixed in #4 and released.
Please try the latest version (`pip install --upgrade gpt4all-j`) and let me know if it fixed your issue.

**Comment by TheBloke** [comment]
Yes thank you, that fixes the problem!
